packages/@react-stately/data/src/useAsyncList.ts

```diff
@@ -200,12 +202,15 @@ function reducer<T, C>(data: AsyncListState<T, C>, action: Action<T, C>): AsyncL
     case 'loadingMore':
       switch (action.type) {
         case 'success':
+          selectedKeys = (data.selectedKeys === 'all' || action.selectedKeys === 'all')
+            ? 'all'
+            : [...data.selectedKeys, ...(action.selectedKeys ?? [])];
```

Comment:
missing `new Set`?

Reply:
I have the set on 213, but I can update per Daniel's comment